File: easy/3798_largest_even_number/solution.py
from collections import deque


class Solution:
    def largestEven(self, s: str) -> str:
        s_deque = deque(s)

        while s_deque and s_deque[-1] != "2":
            if s_deque[-1] == "1":
                s_deque.pop()
            else:
                s_deque.popleft()

        return "".join(s_deque)

File: easy/3798_largest_even_number/solution.cpp
#include <deque>
#include <string>

class Solution {
public:
  std::string largestEven(std::string s) {
    std::deque<char> sDeque(s.begin(), s.end());

    while (!sDeque.empty() && sDeque.back() != '2') {
      if (sDeque.back() == '1')
        sDeque.pop_back();
      else
        sDeque.pop_front();
    }

    return std::string(sDeque.begin(), sDeque.end());
  }
};
